### ISSUE TITLE
Implement water-steam model in PorousFlow

### ISSUE THREAD

**Issue by cpgr** [idaholab/moose#13108]
## Reason
A water-steam EOS that allows for phase change should be added to PorousFlow.

## Design
It can be implemented using the design that is used for the much more complicated brine-CO2 and water-NCG models that are already available.

## Impact
New capability that will be useful for geothermal modelling etc that can use all of the capability that is already in PorousFlow (such as stabilisation, all of the constitutive models etc).

**Comment by WilkAndy** [comment]
Yea, this is good.  I think i did this a while ago while looking at variable switching.  The variable switching was a disaster for convergence, so i left it alone.  Are you planning on doing this Issue, or are you wishing someone else would do it (or both!)?

**Comment by cpgr** [comment]
I've done it already! Just making some examples and documentation

**Comment by WilkAndy** [comment]
Woohoo!

**Comment by WilkAndy** [comment]
I was just mucking around with git and noticed i had a "variable switching" branch that is super old.  But if it's of any interest here, i can send it to you.

**Comment by cpgr** [comment]
Can you send me a link for it?

**Comment by WilkAndy** [comment]
Here's my commentary on some work that i did with variable switching with water-steam.  I found nonlinear convergence was poor:

"Unfortunately not all the Jacobian entries are possible to get right in this example.  I discussed some of this with Chris (it's too long to write the whole story here), but there are also more problems within elements that are "mixed" (some notes are porepressure, some are saturation).  This means quite crappy convergence - takes about 20-times more nonlinear iterations in this particular example that i'm using.

If you're interested, the code is at https://github.com/WilkAndy/moose/tree/variable_switching​

It may be possible to ameliorate the problems by using a different capillary-pressure relationship in this particular example.  Also, it's possible that the problems are particularly severe in this particular example.

Anyway, the important thing is that I've showed our architecture CAN support variable switching with ease!"

I'm sure you've done a better job, Chris!
